outplacement_invoice/models/sale.py

```
@@ -112,41 +96,75 @@ def create_invoice(self, invoice):
                 return invoice
             party = sf_dict['Invoice']['cac:BuyerParty']['cac:Party']
             # Find or create a res_partner for buyer (AF).
-            org_nr = party['cac:PartyIdentification']['cac:ID']
-            res_partner = self.env['res.partner'].search(
-                [('company_registry', '=', org_nr)], limit=1)
+            partner_ext_id = party['cac:PartyIdentification']['cac:ID']['#text']
+            try:
+                res_partner = self.env.ref("__af_data__." + partner_ext_id)
+            except:
```

Do not use a bare except here, capture the explict error you are expecting.